tests/platform/test_platform_info.py

```diff
@@ -259,3 +296,203 @@ def test_show_platform_syseeprom(testbed_devices):
+
+    logging.info("Create PSU controller for testing")
+    psu_ctrl = psu_controller(dut.hostname, dut.facts["asic_type"])
+    if psu_ctrl is None:
+        pytest.skip("No PSU controller for %s, skip rest of the testing in this case" % dut.hostname)
+
+    logging.info("To avoid DUT losing power, need to turn on PSUs that are not powered")
+    turn_all_psu_on(psu_ctrl)
+
+    logging.info("Initialize test results")
+    psu_test_results = {}
+    if not check_all_psu_on(dut, psu_test_results):
+        pytest.skip("Some PSU are still down, skip rest of the testing in this case")
+
+    with ThermalPolicyFileContext(dut, THERMAL_POLICY_VALID_FILE):
+        fan_mocker = mocker_factory(dut, 'FanStatusMocker')
+        if fan_mocker is None:
+            pytest.skip("No FanStatusMocker for %s, skip rest of the testing in this case" % dut.facts['asic_type'])
+
+        fan_mocker.mock_data()  # make data random
+        restart_thermal_control_daemon(dut)
+        wait_until(THERMAL_CONTROL_TEST_WAIT_TIME, THERMAL_CONTROL_TEST_CHECK_INTERVAL, fan_mocker.check_all_fan_speed,
+                   60)
+
+        check_thermal_algorithm_status(dut, mocker_factory, False)
+
+        all_psu_status = psu_ctrl.get_psu_status()
+        psu = all_psu_status[0]
+        turn_off_psu_and_check_thermal_control(dut, psu_ctrl, psu, fan_mocker)
+        psu_test_results.clear()
+        if not check_all_psu_on(dut, psu_test_results):
+            pytest.skip("Some PSU are still down, skip rest of the testing in this case")
+
+        psu = all_psu_status[1]
+        turn_off_psu_and_check_thermal_control(dut, psu_ctrl, psu, fan_mocker)
+        psu_test_results.clear()
+        if not check_all_psu_on(dut, psu_test_results):
+            pytest.skip("Some PSU are still down, skip rest of the testing in this case")
+
+        wait_until(THERMAL_CONTROL_TEST_WAIT_TIME, THERMAL_CONTROL_TEST_CHECK_INTERVAL, fan_mocker.check_all_fan_speed,
+                   65)
+
+
+def turn_off_psu_and_check_thermal_control(dut, psu_ctrl, psu, mocker):
+    """
+    @summary: Turn off PSUs, check all FAN speed are set to 100% according to thermal
+              control policy file.
+    """
+    logging.info("Turn off PSU %s" % str(psu["psu_id"]))
+    psu_ctrl.turn_off_psu(psu["psu_id"])
+    time.sleep(5)
+
+    psu_under_test = None
+    psu_line_pattern = re.compile(r"PSU\s+\d+\s+(OK|NOT OK|NOT PRESENT)")
+    cli_psu_status = dut.command(CMD_PLATFORM_PSUSTATUS)
+    for line in cli_psu_status["stdout_lines"][2:]:
+        assert psu_line_pattern.match(line), "Unexpected PSU status output"
+        fields = line.split()
+        if fields[2] != "OK":
+            psu_under_test = fields[1]
+
+    assert psu_under_test is not None, "No PSU is turned off"
+    wait_until(THERMAL_CONTROL_TEST_WAIT_TIME, THERMAL_CONTROL_TEST_CHECK_INTERVAL, mocker.check_all_fan_speed, 100)
+
+    psu_ctrl.turn_on_psu(psu["psu_id"])
+
```

Comment:
need to check PSU status, if not all PSU up should raise warning/error?

Reply:
It will raise a warning outside this function.